pkg/assembler/helpers/purl.go

```diff
@@ -23,11 +23,13 @@ import (
 	"strings"
 
 	model "github.com/guacsec/guac/pkg/assembler/clients/generated"
+	cpecommon "github.com/knqyf263/go-cpe/common"
 	purl "github.com/package-url/packageurl-go"
 )
 
 const (
 	PurlTypeGuac  = "guac"
+	PurlTypeCPE   = "cpe"
```

Comment:
I'm hesitant to have CPE as a pkg trie since it is a bit more like a regex expression than a descriptor of a package, my initial thought is to perhaps have this as a `MetadataFor` package node

Reply:
@pxp928 @mihaimaruseac @mlieberman85 @jeffmendoza thoughts?

Reply:
Having it as a `MetadataFor` would reduce the overall redundant information and make traversal for information easier

Reply:
> Having it as a `MetadataFor` would reduce the overall redundant information and make traversal for information easier

Does `MetadataFor` mean `HasMetadata`?
Or are suggesting to add a new `MetadataFor` node?

Reply:
sorry yes [HasMetadata](https://github.com/guacsec/guac/blob/main/pkg/assembler/graphql/schema/metadata.graphql#L36)

Reply:
> sorry yes [HasMetadata](https://github.com/guacsec/guac/blob/main/pkg/assembler/graphql/schema/metadata.graphql#L36)

Cool, thanks for confirming :+1:

Reply:
I also think it makes sense to do this via `HasMetadata`. If we discover that this is not easily searchable and things would improve when using CPE trie, we can switch and take in the cost of duplicating the information

Reply:
Sorry guys, regarding this duplication topic, let me try to summarize the 2 approaches.

"CPE Package" approach in this PR:

- ingests 1 package (CPE based) and 2 edges (`PkgEqual`, `IsDependency`)
- retrieves the relation between CPE based pkg and dependency pkg with 1 `IsDependency` request

"HasMetadata" approach:

- ingests 1 Metadata (CPE based) and 1 edge (`HasMetadata`)
- retrieves the relation between CPE "metadated" pkg and dependency pkg with 2 requests: `HasMetadata` and `IsDependency`

So, hoping I didn't miss anything, both approaches add 1 Node but the former add 1 more edge and the former requires one more traversal to retrieve the same information.

TBH, I feel like adding one more edge is a better approach rather than having to execute 2 traversals for getting the same information.

WDYT?

Reply:
The main difference is there would be no need to create the pkg trie for the CPE (as they will be heuristic-based). The `HasMetadata` would just be one node attached to the package